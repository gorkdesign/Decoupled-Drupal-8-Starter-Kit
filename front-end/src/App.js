import React from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import { Query } from 'react-apollo';

import { ErrorBoundary } from './utils/ErrorBoundary';
import PrivateRoute from './routes/PrivateRoute';
import Logout from './routes/Logout';

import PageTemplate from './containers/PageTemplate';
import Login from './containers/LoginPage';
import Home from './containers/HomePage';
import List from './containers/ListPage';
import Edit from './containers/EditPage';
import ContentDetailsPage from './containers/ContentDetailsPage';

import { SESSION_QUERY } from './api/apolloProxy';
// import { defaults as defaultSession } from './api/resolvers';

export const {
  Provider: SessionProvider,
  Consumer: SessionConsumer,
} = React.createContext();

const App = () =>
  (
    <ErrorBoundary>
      <Query query={SESSION_QUERY} notifyOnNetworkStatusChange>
        {
          ({
            loading, error, data, networkStatus,
          }) => {
            // console.log(`networkStatus: ${networkStatus}`);
            if (networkStatus === 4) return 'Refetching!';
            if (networkStatus === 8) return 'NETWORK ERROR!';
            if (loading) return 'Loading!';
            if (error) return `Error!: ${error}`;

            return (
              <SessionProvider value={{ ...data.session }}>
                <Router>
                  <Switch>
                    <PageTemplate>
                      <Route path="/" exact component={Home} />
                      <Route path="/articles/:path" exact component={ContentDetailsPage} />
                      <Route path="/list" exact component={List} />
                      <PrivateRoute path="/edit" component={Edit} />
                      <Route path="/login" exact component={Login} />
                      <Route path="/logout" exact component={Logout()} />
                    </PageTemplate>
                  </Switch>
                </Router>
              </SessionProvider>
            );
          }
        }
      </Query>
    </ErrorBoundary>
  );

export default App;
