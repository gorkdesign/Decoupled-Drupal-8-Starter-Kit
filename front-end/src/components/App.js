import React, {Component} from 'react';

import { ApolloProvider } from 'react-apollo';

import NodeEditPage from 'containers/NodeEditPage';

import CardList from 'components/CardList';
import SignIn from 'components/SignIn';
import Navbar from 'components/Navbar';

import { connect } from 'react-redux';

export class App extends Component {

  renderLoading = () => {
    return (
      <div>Loading...</div>
    );
  }

  renderAuthenticated = () => {

    const { handleLogout, apolloclient, activeNode } = this.props;

    return (
      <div>
        {apolloclient ?
          <ApolloProvider client={apolloclient}>
            <div className="container authenticated">
                <Navbar handleLogout={handleLogout} />
                {
                  !this.props.activeNode ?
                    <CardList projectCardListHandler={this.props.projectCardListHandler}/>
                  :
                    <NodeEditPage activeNode={activeNode} />
                }
            </div>
          </ApolloProvider> : ''
        }
      </div>

    );
  }

  renderAnonymous = () => {
    const { handleInputChange, handleLogin, isLoginFailed } = this.props;
    return <SignIn handleInputChange={ handleInputChange } handleLogin={ handleLogin } isLoginFailed={ isLoginFailed } />;
  }

  renderError = () => {
    return (
      <div>Sadly, there seems to have been an error. Contact someone super important to help facilitate forward progress.</div>
    )
  }

  render() {
    if (this.props.isLoading) {
      return this.renderLoading();
    } else if (this.props.authenticated){
      return this.renderAuthenticated();
    } else if (!this.props.authenticated && !this.props.isLoading) {
      return this.renderAnonymous();
    } else {
      return this.renderError();
    }
  }

}
const mapStateToProps = (state, ownProps) => ({
  authenticated: state.oauth.authenticated,
  apolloclient: state.apollo.apolloClient
})

export default connect(mapStateToProps)(App);
