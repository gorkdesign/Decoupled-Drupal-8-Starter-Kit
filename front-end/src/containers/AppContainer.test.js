import React from 'react';
import { mount, shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import createSagaMiddleware from 'redux-saga';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';

import App from './AppContainer';


describe('AppContainer', () => {
  const sagaMiddleware = createSagaMiddleware();
  const mockStore = configureMockStore([sagaMiddleware]);
  const store = mockStore({});
  const initialPropsState = {
    activeArticleNid: 0,
    articles: [],
  };

  let props;
  let mountedLogin;
  const AppContainer = (modifiedProps = initialPropsState) => {
    props = {
      ...initialPropsState,
      ...modifiedProps,
    };

    if (!mountedLogin) {
      mountedLogin = mount(<App {...props} />);
    }
    return mountedLogin;
  };

  beforeEach(() => {
    props = initialPropsState;
    mountedLogin = undefined;
  });

  describe('Snapshots', () => {
    const output = shallow(<Provider store={store}><AppContainer {...initialPropsState} /></Provider>);
    it('should render correctly', () => {
      expect(shallowToJson(output)).toMatchSnapshot();
    });
  });


  // describe('AppContainerArticle differing props', () => {
  //   it('Should NOT display the login form if authenticated - SHOULD REDIRECT', () => {
  //     const newProps = { ...initialPropsState };
  //     newProps.isAuthenticated = true;
  //     const wrapper = shallow(<Login {...newProps} />);
  //     expect(wrapper.find(Redirect).props().to).toBe('/');
  //   });
  // });
});
