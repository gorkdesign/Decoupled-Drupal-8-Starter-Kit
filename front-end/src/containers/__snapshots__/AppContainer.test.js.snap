// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppContainer Snapshots should render correctly 1`] = `
<AppContainer
  activeArticleNid={0}
  articles={Array []}
/>
`;
